src/libraries/Common/tests/System/Net/RandomReadWriteSizeStream.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Test.Common
+{
+    // Wrapper stream to artificially chop reads and writes to smaller chunks.
+    public class RandomReadWriteSizeStream : Stream
+    {
+        private readonly Stream _innerStream;
+        private readonly int _maxSize;
+
+        public RandomReadWriteSizeStream(Stream stream, int maximumSize = int.MaxValue)
+        {
+            _innerStream = stream;
+            _maxSize = maximumSize;
+        }
+
+        public override bool CanSeek => false;
+
+        public override bool CanRead => _innerStream.CanRead;
+
+        public override bool CanTimeout => _innerStream.CanTimeout;
+
+        public override bool CanWrite => _innerStream.CanWrite;
+
+        public override long Position
```

Comment:
Since this is a non-seekable stream we should probably just throw for Position, Seek, Length, and SetLength.

Reply:
There is really need to make it non-seekable since the read/write really does not care. So I updated `CanSeek` to defer to underlying stream as well to make everything consistent.

Reply:
OK